shell/src/main/java/alluxio/cli/fs/command/FreeCommand.java

```diff
@@ -64,6 +68,15 @@ protected void runPlainPath(AlluxioURI path, CommandLine cl)
       throws AlluxioException, IOException {
     FreeOptions options = FreeOptions.defaults().setRecursive(true).setForced(cl.hasOption("f"));
     mFileSystem.free(path, options);
+    CommonUtils.waitFor("File to be freed.", (v) -> {
```

Comment:
There's a race condition where a reader might load part of the file immediately after it is freed - to account for this we could do a couple of retries with calling `free`, and if we time out we can mention that another reader might be loading the file as we try to free it

Reply:
Updated, thanks.